demo-plugin/src/main/kotlin/org/jetbrains/research/refactoringDemoPlugin/modelInference/LicenseDetector.kt

```diff
@@ -0,0 +1,95 @@
+package org.jetbrains.research.refactoringDemoPlugin.modelInference
+
+import io.kinference.data.map.ONNXMap
+import io.kinference.data.seq.ONNXSequence
+import io.kinference.data.tensors.Tensor
+import io.kinference.data.tensors.asTensor
+import io.kinference.model.Model
+import io.kinference.ndarray.arrays.FloatNDArray
+import io.kinference.ndarray.arrays.LongNDArray
+import java.util.Locale
+import org.jetbrains.research.refactoringDemoPlugin.modelInference.license.Apache_2_0
+import org.jetbrains.research.refactoringDemoPlugin.modelInference.license.BSD_3_Clause
+import org.jetbrains.research.refactoringDemoPlugin.modelInference.license.License
+import org.jetbrains.research.refactoringDemoPlugin.modelInference.license.MIT
+import kotlin.time.ExperimentalTime
+
+/**
+ * See [Sorrel plugin source code](https://github.com/JetBrains-Research/sorrel).
+ */
+class LicenseDetector {
+    // Classes for decode numeric predictions
+    private val classes: List<String> = LicenseDetector::class.java.getResourceAsStream(
+        "/model/license_level_classes_v2.txt"
+    ).reader().readLines()
+
+    // Detected License (string) to License class mapping
+    private val licenseToClass = mapOf(
+        "Apache-2.0" to Apache_2_0,
+        "MIT" to MIT,
+        "BSD-3-Clause" to BSD_3_Clause,
+    )
+
+    // Model & vectorizer for detection licenses on project level initializiation
```

Comment:
detection __of__ licenses or detect**ing** licenses

Reply:
Ooops, the code was taken from the Sorrel plugin. I will check the comments more carefully. Thanks!